homeassistant/components/linear_garage_door/light.py

```diff
@@ -0,0 +1,133 @@
+"""Linear garage door light."""
+
+from datetime import timedelta
+import math
+from typing import Any
+
+from linear_garage_door import Linear
+
+from homeassistant.components.light import ATTR_BRIGHTNESS, ColorMode, LightEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import LinearUpdateCoordinator
+
+SCAN_INTERVAL = timedelta(seconds=60)
+PARALLEL_UPDATES = 0
+SUPPORTED_SUBDEVICES = ["Light"]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Linear Garage Door cover."""
+    coordinator: LinearUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    data = coordinator.data
+
+    async_add_entities(
+        LinearLightEntity(
+            device_id=device_id,
+            device_name=data[device_id]["name"],
+            subdevice=subdev,
+            config_entry=config_entry,
+            coordinator=coordinator,
+        )
+        for device_id in data
+        for subdev in data[device_id]["subdevices"]
+        if subdev in SUPPORTED_SUBDEVICES
+    )
+
+
+class LinearLightEntity(CoordinatorEntity[LinearUpdateCoordinator], LightEntity):
+    """Light for Linear devices."""
+
+    _attr_color_mode = ColorMode.BRIGHTNESS
+    _attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+
+    def __init__(
+        self,
+        device_id: str,
+        device_name: str,
+        subdevice: str,
+        config_entry: ConfigEntry,
+        coordinator: LinearUpdateCoordinator,
+    ) -> None:
+        """Initialize the light."""
+        super().__init__(coordinator)
+
+        self._attr_has_entity_name = True
+        self._attr_name = subdevice
+        self._attr_unique_id = f"{device_id}-{subdevice}"
+        self._config_entry = config_entry
+        self._device_id = device_id
+        self._subdevice = subdevice
+        self.device_info = DeviceInfo(
+            identifiers={(DOMAIN, self._device_id)},
+            name=device_name,
+            manufacturer="Linear",
+            model="Garage Door Opener",
+        )
+
```

Comment:
I'm assuming this is also present in cover.py. consider creating a base entity where all similar calls are stored to reduce code duplication